### PR TITLE
Zadanie 2

### PR DESCRIPTION
**Pull request by nowikarol**

### CODE REVIEW

**Review by Kukann25 — COMMENTED** [review]
Dobrze by było zmierzyć więcej metryk niż samo accuracy. Precision recall i F1 mogą okazać się bardziej pomocne.

**Review by wsparcie — APPROVED** [review]
spoko wnioski z poprzedniego zadania i fajne wyjasnienie wyboru metryk 
fajne wykorzystanie stratify=y przy podziale zbioru
trafny pomysl z uzyciem robustscaler w obliczu outlierow w Age i Fare spoko uzycie dummy classifier jako baselineu
dobre przeszukanie hiperparametrow przez gridsearchcv i fajne mapowanie kolumn z wynikami 
bardzo czytelne i ladne wykresy zaleznosci dokladnosci od hiperparametrow 
w gridSsearchcv scoring=accuracy mozna wsm zamienic na f1 albo uzyc multi-metric scoring, zwlaszcza ze trafnie o tym F1 wspomniano we wnioskach :)
przy optymalizacji xgboost siatka jest troche skromna, mozna dodac np. min child weight albo gamma, zeby dodatkowo zminimalizowac ryzyko overfittingu

lgtm

**Review by TsuneKioo — COMMENTED** [review]
Fajnie, że wyjaśniłeś dlaczego używasz accuracy i dobrze że zwracasz uwagę też na F1. 

Podoba mi się porównanie wszystkich hiperparametrów na jednym wykresie. Łatwo sprawdzić, która kombinacja jest wygrywająca.

**Review by omakszewska — CHANGES_REQUESTED** [review]
Niestety wkradł się wyciek danych:
-  RObustScaler na całym X train, walidacja krzyżowa dosatje juz przeskalowane dane z całego X_train. Rozwiązaniem byłoby dodanie scalera do Pipeline przekazywanym do GridSearchCV

Raport na końcu pokazuje więcej metryk, ale w trakcie wyboru metryk kierujesz się tylko accuracy, lepiej dodać multi-metric

Za płytka eksploracja hiperparametrów dla XGBoost (chociażby min_child_weight, gamma)

Za krótka analiza, warto się trochę tym pobawić

Na plus porównanie modeli i końcowy raport.
Potrzebne poprawki, glównie Pipeline, bogatsze strojenie + ewaluacja XGBoosta
